python/taichi/snode/fields_builder.py

```diff
@@ -36,15 +36,26 @@ class FieldsBuilder:
     """
     _finalized_fbs = []
 
-    def __init__(self):
-        self._ptr = _snode_registry.create_root()
-        self._root = snode.SNode(self._ptr)
+    def __init__(self, ptr=None, root=None):
+        if ptr is None:
+            self._ptr = _snode_registry.create_root()
+        else:
+            self._ptr = ptr
+        if root is None:
+            self._root = snode.SNode(self._ptr)
+        else:
+            self._root = root
         self._finalized = False
         self._empty = True
 
     @classmethod
     def finalized_fbs(cls):
```

Comment:
Maybe we can change finalized_fbs() to finalized_roots()?

Reply:
There are only two places that are using finalized_fbs():
https://github.com/taichi-dev/taichi/blob/cd96133e9e45df588c53042099a0b8b37b6c75e1/tests/python/test_tensor_reflection.py#L62
https://github.com/taichi-dev/taichi/blob/cd96133e9e45df588c53042099a0b8b37b6c75e1/python/taichi/lang/__init__.py#L408-L409

In fact, in these places, all we need is the root of SNodeTrees. So, it will have the same effect if we only save the SNodes.

Reply:
That's how I thought about it in the beginning. However, there are also places that uses self._ptr.